lib/takeaway.rb

```diff
@@ -0,0 +1,41 @@
+# require_relative 'twilio'
+
+class Takeaway
+
+  attr_accessor :list, :order, :amount_due, :confirmation
+
+  def initialize
+    @list = {'red curry' => 5, 'green curry' => 5}
+    @order = []
+    @amount_due = 0
+    @confirmation = ""
+  end
+
+  def register_order (item)
+    order << item
+    calculate_order
+  end
+
+  def calculate_order
+    fail 'no order placed' if empty?
+    @amount_due = order.map {|x| x.values}.flatten(1).inject(:+)
```

Comment:
Space between { and | missing.<br>Space missing inside }.<br>Pass `&:values` as an argument to `map` instead of a block.